portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,43 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        
+        List<Entity> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            comments.add(entity);
+        }
+        
+        for (int i = 0; i < comments.size(); i++) {
+            Key commentEntityKey = comments.get(i).getKey();
+            datastore.delete(commentEntityKey);
+        }
```

Comment:
I feel like this doesn't need an extra `ArrayList` and could just be done in one loop instead of two. Let me know if I misunderstood something about the `PreparedQuery`.

Reply:
I thought I was deleting elements from the same thing as I was iterating and that's why I thought couldn't put in the same loop. Will change this. thanks!

Reply:
Yeah this is where I'm not 100% sure. I can't quite tell from the documentation on `PreparedQuery` and `DatastoreService` since you are technically iterating over the query and deleting from the datastore. This may be a question for the technical support chat, but at least `PreparedQuery` has an `asList` method that will do what your first loop does.